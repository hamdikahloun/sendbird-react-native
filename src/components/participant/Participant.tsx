import React from 'react';
import {StyleSheet, Text, TouchableOpacity} from 'react-native';
import {User} from 'sendbird';
import {Avatar} from '../avatar';

type ParticipantProps = {
  participant: User;
  onParticipantPressed?: (participant: User) => void;
};
export const Participant = ({
  participant,
  onParticipantPressed,
}: ParticipantProps) => {
  const onPress = () => {
    onParticipantPressed?.(participant);
  };
  return (
    <TouchableOpacity
      disabled={!onParticipantPressed}
      onPress={onPress}
      style={styles.container}>
      <Avatar image={participant.profileUrl} />
      <Text style={styles.text}>{participant.nickname}</Text>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  container: {
    backgroundColor: '#F6F6F6',
    borderRadius: 16,
    justifyContent: 'center',
    alignItems: 'center',
    padding: 8,
    borderWidth: 1,
    borderColor: 'white',
    marginEnd: 16,
  },
  text: {
    color: '#0C0020',
    fontSize: 10,
  },
});
