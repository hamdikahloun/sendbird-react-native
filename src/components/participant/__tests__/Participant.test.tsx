import React from 'react';
import {render} from '@/../testing';
import {Participant} from '../Participant';

describe('Participant', () => {
  it('should render correctly', async () => {
    const participant: any = {nickname: 'Hamdi'};
    const screen = render(<Participant participant={participant} />).toJSON();
    expect(screen).toMatchSnapshot();
  });
});
