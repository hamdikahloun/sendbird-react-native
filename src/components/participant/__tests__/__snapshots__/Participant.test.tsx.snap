// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Participant should render correctly 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F6F6F6",
      "borderColor": "white",
      "borderRadius": 16,
      "borderWidth": 1,
      "justifyContent": "center",
      "marginEnd": 16,
      "opacity": 1,
      "padding": 8,
    }
  }
>
  <View
    accessibilityState={
      Object {
        "disabled": true,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 24,
            "height": 48,
            "width": 48,
          },
          undefined,
        ]
      }
    >
      <Image
        onError={[Function]}
        source={
          Object {
            "uri": "https://avatars.githubusercontent.com/u/5143408?s=200&v=4",
          }
        }
        style={
          Array [
            Object {
              "borderRadius": 24,
              "height": 48,
              "width": 48,
            },
            undefined,
          ]
        }
        testID="user-avatar"
      />
    </View>
  </View>
  <Text
    style={
      Object {
        "color": "#0C0020",
        "fontSize": 10,
      }
    }
  >
    Hamdi
  </Text>
</View>
`;
